src/index/vocabulary/PolymorphicVocabulary.h

```diff
@@ -87,6 +88,39 @@ class PolymorphicVocabulary {
         },
         vocab_);
   }
+
+  // Analogous to `lower_bound`, but since `word` is guaranteed to be a full
+  // word, not a prefix, this function can respect the split of an underlying
+  // `SplitVocabulary`.
+  template <bool getUpperBound, typename String, typename Comp>
+  WordAndIndex getPositionOfWord(const String& word, Comp comp) const {
+    return std::visit(
+        [&word, &comp](auto& vocab) {
+          using T = std::decay_t<decltype(vocab)>;
+          if constexpr (ad_utility::isInstantiation<T, SplitVocabulary>) {
+            return vocab
+                .template getPositionOfWord<String, Comp, getUpperBound>(
+                    word, std::move(comp));
+          }
+          if constexpr (getUpperBound) {
+            return vocab.upper_bound(word, std::move(comp));
+          } else {
+            return vocab.lower_bound(word, std::move(comp));
+          }
+        },
+        vocab_);
+  }
+
+  template <typename String, typename Comp>
+  WordAndIndex getPositionOfWordLower(const String& word, Comp comp) const {
+    return getPositionOfWord<false>(word, comp);
+  };
+
+  template <typename String, typename Comp>
+  WordAndIndex getPositionOfWordUpper(const String& word, Comp comp) const {
+    return getPositionOfWord<true>(word, comp);
+  };
+
```

Comment:
See what I wrote above, this should be a single function that calls lowerBound and then just checks the word that was found at the position.